src/clients/QBittorrent.ts

```diff
@@ -296,84 +326,88 @@ export default class QBittorrent implements TorrentClient {
 					}
-				: await this.getTorrentConfiguration(searchee);
-
-			const newCategoryName =
-				duplicateCategories && searchee.infoHash
-					? await this.setUpCrossSeedCategory(category)
-					: category;
-
+				: {
+						savePath: searcheeInfo!.save_path,
+						isComplete: this.isTorrentComplete(searcheeInfo!),
+						autoTMM: searcheeInfo!.auto_tmm,
+						category: searcheeInfo!.category,
+					};
 			if (!isComplete) return InjectionResult.TORRENT_NOT_COMPLETE;
+			const filename = `${newTorrent.getFileSystemSafeName()}.${TORRENT_TAG}.torrent`;
+			const buffer = new Blob([newTorrent.encode()], {
+				type: "application/x-bittorrent",
+			});
+			const toRecheck = shouldRecheck(decision);
 
-			const contentLayout =
-				!path &&
-				newTorrent.isSingleFileTorrent &&
-				(await this.isSubfolderContentLayout(searchee))
-					? "Subfolder"
-					: "Original";
-
-			const skipRecheckTorrent =
-				decision === Decision.MATCH_PARTIAL ? skipRecheck : true;
-
+			// ---------------------- Building form data ----------------------
 			const formData = new FormData();
 			formData.append("torrents", buffer, filename);
-			formData.append("tags", TORRENT_TAG);
-			formData.append("category", newCategoryName);
-
-			if (autoTMM) {
-				formData.append("autoTMM", "true");
-			} else {
-				formData.append("autoTMM", "false");
-				formData.append("savepath", save_path);
-			}
 			if (path) {
-				formData.append("contentLayout", "Original");
-				formData.append("skip_checking", skipRecheck.toString());
-				formData.append("paused", (!skipRecheck).toString());
-			} else {
-				formData.append("contentLayout", contentLayout);
-				formData.append("skip_checking", skipRecheckTorrent.toString());
-				formData.append("paused", (!skipRecheckTorrent).toString());
+				formData.append("savepath", savePath);
 			}
-
+			formData.append(
+				"autoTMM",
+				flatLinking && searchee.infoHash ? autoTMM.toString() : "false",
+			);
+			formData.append("category", category);
+			formData.append(
+				"tags",
+				this.getTagsForNewTorrent(searchee, searcheeInfo, category),
+			);
+			formData.append(
+				"contentLayout",
+				this.getLayoutForNewTorrent(searchee, searcheeInfo, path),
+			);
+			formData.append("skip_checking", !toRecheck.toString());
+			formData.append("paused", toRecheck.toString());
 			// for some reason the parser parses the last kv pair incorrectly
 			// it concats the value and the sentinel
 			formData.append("foo", "bar");
+			await this.addTorrent(formData);
 
-			await this.request("/torrents/add", formData);
-
-			if (path && !skipRecheck) {
-				await new Promise((resolve) => setTimeout(resolve, 100));
-				await this.request(
-					"/torrents/recheck",
-					`hashes=${newTorrent.infoHash}`,
-					X_WWW_FORM_URLENCODED,
+			await wait(1000);
+			const newInfo = await this.getTorrentInfo(newTorrent.infoHash);
+			if (!newInfo) {
+				throw new CrossSeedError(
+					`Failed to retrieve torrent after adding`,
```

Comment:
CrossSeedError is admittedly very poorly named, but is intended to represent errors that we are certain are misconfigurations. In this case we should just use Error.